### ISSUE TITLE
Keyboard doesn't open - Disabled animations

### ISSUE THREAD

**Issue by desmonk** [TeamNewPipe/NewPipe#7647]
### Checklist

- [x] I am using the latest version - 0.21.15 
- [x] I checked, found duplicates (open OR closed) of this issue in the repo.  #7607 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one bug. I will open one issue for every bug report I want to file.

### Steps to reproduce the bug

0. ``Dev note: Animations have to be disabled - otherwise the bug is not reproducible``
1. Pull down a video to the bottom of the player as seen below.
2. Search bar won't call they keyboard even after multiple clicks.
3. Pull up the video to the normal view and pull it down again.
4. Search bar now will call the keyboard as intended.

### Actual behavior

Search fails to call the keyboard.

### Expected behavior

Search should call the keyboard.

### Screenshots/Screen recordings

![IMG_000023](https://user-images.githubusercontent.com/55331939/149009918-bc32765c-cb14-483e-b657-913097f86576.gif)

### Logs

NA

### Device info

 - Android version/Custom ROM version: 11
 - Device model: OnePlus

### Note
Tested https://github.com/TeamNewPipe/NewPipe/suites/4810100849/artifacts/135282702 found at #7607 but didn't work.

**Comment by litetex** [comment]
I try to reproduce it but I'm unable to do so:

https://user-images.githubusercontent.com/40789489/149392717-1c447a56-96a6-4ecb-8a20-1ad9adc9aff0.mp4

**Comment by desmonk** [comment]
Difference here is the empty search bar, unable to call the keyboard.

**Comment by AndRo-Marian** [comment]
I see this happening when you opn the app first time after you close the instance.

**Comment by Darkmaster006** [comment]
I have this problem too. Galaxy A9, with Android 8 (latest one it can get). The first time you tap the 'Search' bar, it justs doesn't show the keyboard. You gotta tap the 'x' then tap it again, and it will show it.

**Comment by dtcxzyw** [comment]
Hi @litetex, I would like to work on this issue. The bug is reproduced on the latest dev branch. SearchFragment tries to open the soft keyboard before acquiring the focus. But the request is rejected by InputMethodManager.

**Comment by litetex** [comment]
So I had a look into the problem (when animations are turned off).
Looks like a bug inside the Android framework itself:
When the keyboard is requested ``showSoftInput`` is called:
![grafik](https://user-images.githubusercontent.com/40789489/163728408-3a037b5f-ba87-4c54-b734-016abce3e030.png)
However when ``hasServedByInputMethodLocked`` is called it returns ``false`` because no ``servedView`` could be found.
This is caused by a race condition:

When animations are enabled and another view has foces the following happens:
Before the keyboard is opened this is called which set's the ``servedView``:
![grafik](https://user-images.githubusercontent.com/40789489/163729354-c8f581a0-29a8-4010-9fa5-79ce795a5207.png)

However when animations are disabled, this is not done and Android falls back to no served view:
![grafik](https://user-images.githubusercontent.com/40789489/163729436-cb78d4ab-166b-4a35-a0fa-0d074df1800e.png)

and this in turn is only done because ``mNextServedView`` was set to ``null`` before:
![grafik](https://user-images.githubusercontent.com/40789489/163729754-30a0964b-de8e-434f-9f00-da7baf992f7a.png)
![grafik](https://user-images.githubusercontent.com/40789489/163729961-24670e1e-dec0-48b4-8326-9fd966a26f53.png)

TL;DR
* The list with videos (item_list) has focus
* The search button is clicked
* The focus is set to the editTextField
* ``ImeFocusController`` get's this as ``mNextServedView``
* Because animations are disabled ``ImeFocusController``'s ``mNextServedView`` is immediately set to ``null`` inside ``onViewDetachedFromWindow`` (this is caused by the item_list losing focus)
![grafik](https://user-images.githubusercontent.com/40789489/163730377-a53a5282-665e-4525-ab46-91d84f65288c.png)
  * With active animations this is done later,  ``mServedView`` differs from the ``view`` parameter and ``mNextServedView`` is never set to ``null`` → no problem occurs
* This causes the ``mServedView`` to also become ``null``
* ... which fails to show the search
